api/grpcserver/grpcserver_test.go

```diff
@@ -114,10 +116,33 @@ var (
 	stateRoot = types.HexToHash32("11111")
 )
 
-func init() {
+func TestMain(m *testing.M) {
 	// run on a random port
 	cfg.GrpcServerPort = 1024 + rand.Intn(9999)
 
+	atx := types.NewActivationTx(challenge, addr1, nipost, numUnits, nil)
+	if err := activation.SignAtx(signer, atx); err != nil {
+		log.Println("failed to sign atx:", err)
+		os.Exit(1)
+	}
+	var err error
+	globalAtx, err = atx.Verify(0, 1)
+	if err != nil {
+		log.Println("failed to verify atx", err)
```

Comment:
does the err needs formatting?

Reply:
TIL: `fmt.Println` (and `log` is just an extension of that) will not only check if an object implements the `fmt.Stringer` interface, but also `error` and automatically call `err.Error()` to get the string

proof: https://go.dev/play/p/Jd5JrIsVVQc